plugins/diffview-resources/src/components/Highlight.svelte

```diff
@@ -0,0 +1,24 @@
+<!--
+// Copyright © 2024 Hardcore Engineering Inc.
+//
+// Licensed under the Eclipse Public License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may
+// obtain a copy of the License at https://www.eclipse.org/legal/epl-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+-->
+<script lang="ts">
+  import { highlightText } from '../highlight'
+
+  export let value: string
+  export let language: string | undefined = undefined
+
+  $: highlighted = highlightText(value, { language })
+</script>
+
+{@html highlighted}
```

Comment:
Is it safe to use @html with a value produced from a user input?

Reply:
it is safe here, because we use @html on the input processed by highlight.js

they mention that html produced by the highlighter **is safe**:
https://github.com/highlightjs/highlight.js/issues/3229

We can post-process with something like DOMPurify, but I do not see much sense.

Reply:
Looks good but should we add a couple of tests to verify that the input is sanitized so we have some additional layer of security. E.g. in case some defaults change in `highlight` or something.

Reply:
Added dompurify 😄